zio-cli/shared/src/main/scala/zio/cli/oauth2/OAuth2Provider.scala

```diff
@@ -0,0 +1,266 @@
+package zio
+package cli
+package oauth2
+
+import java.net.URI
+import java.net.http.HttpRequest
+
+import zio.json._
+
+/**
+ * Provider of OAuth2 authorization.
+ */
+sealed trait OAuth2Provider extends Serializable with Product {
```

Comment:
We can probably unseal this so users can implement their own provider, unless you rely on exhaustive pattern matching somewhere?

Reply:
Good suggestion. Unsealed.